File: IDAUtil/Service/DistributionListCalculator.cs
using IDAUtil.Model.Properties.ServerProperty;
using lib;
using Microsoft.VisualBasic;
using System.Data;
using System.Linq;

namespace IDAUtil.Service {
    public class DistributionListCalculator: IDistributionListCalculator {
        private readonly IDBServerConnector dbServer;

        //Constructor
        public DistributionListCalculator(IDBServerConnector dbServer) {
            this.dbServer = dbServer;
            dbServer.createConnectionToServerCF();
        }
        
        //Destructor
        ~DistributionListCalculator() {
            dbServer.closeConnection();
        }

        public string getDistList(string salesOrg, string task) {
            var list = dbServer.getObjectListByQuery<DistributionListProperty>($"SELECT [address] FROM DistributionListData WHERE salesOrg = '{salesOrg}' AND [name] = '{task}'");
            var listArray = list.Select(every => every.address).ToArray();
            return Strings.Join(listArray, ";");
        }
    }
}

File: IDAUtil/Service/DistributionListCalculator.vb
Imports [lib]

Public Class DistributionListCalculator

    Private ReadOnly dbServer As IDBServerConnector

    Public Sub New(dbServer As IDBServerConnector)
        Me.dbServer = dbServer
        dbServer.createConnectionToServerIniCF()
    End Sub

    Protected Overrides Sub Finalize()
        dbServer.closeConnection()
        MyBase.Finalize()
    End Sub

    Public Function getDistList(salesOrg As String, task As String) As String
        Dim list As List(Of DistributionListProperty) = dbServer.getObjectListByQuery(Of DistributionListProperty)($"SELECT [address] FROM DistributionListData WHERE salesOrg = '{salesOrg}' AND [name] = '{task}'")
        Dim listArray As String() = list.Select(Function(every) every.address).ToArray()

        Return Join(listArray, ";")
    End Function
End Class
